express-api/src/services/buildings/buildingServices.ts

```diff
@@ -0,0 +1,121 @@
+import { Building } from '@/typeorm/Entities/Building';
+import { AppDataSource } from '@/appDataSource';
+import { ErrorWithCode } from '@/utilities/customErrors/ErrorWithCode';
+import { DeepPartial, FindOptionsOrder } from 'typeorm';
+import { BuildingFilter } from '@/services/buildings/buildingSchema';
+
+const buildingRepo = AppDataSource.getRepository(Building);
+
+/**
+ * @description          Adds a new building to the datasource.
+ * @param   building     Incoming building data to be added to the database.
+ * @returns {Response}   A 201 status and the building data added.
+ * @throws ErrorWithCode If the building already exists or is unable to be added.
+ */
+export const addBuilding = async (building: Building) => {
+  const existingBuilding = await getBuildingById(building.Id);
+  if (existingBuilding) {
+    throw new ErrorWithCode('Building already exists.', 409);
+  }
+  const newBuilding = buildingRepo.save(building);
+  return newBuilding;
+};
+
+/**
+ * @description Finds and returns a building with matching Id
+ * @param       buildingId  - Number representing building we want to find.
+ * @returns     findBuilding - Building data matching Id passed in.
+ */
+export const getBuildingById = async (buildingId: number) => {
+  try {
+    const findBuilding = await buildingRepo.findOne({
+      where: { Id: buildingId },
+    });
+    return findBuilding;
+  } catch (e) {
+    throw new ErrorWithCode(e.message, e.status);
+  }
```

Comment:
It's ok to not catch anything here. The findOne doesn't throw an error, it just returns null if not found.
```suggestion
  const findBuilding = await buildingRepo.findOne({
    where: { Id: buildingId },
  });
  return findBuilding;
```

Reply:
Yeah, seconding this. I think retrieving a value by id and getting null back is not an exception. It's better to let the caller decide whether this is an error or not since in some cases it probably is expected.